### PR TITLE
Feat: GETEX and GETDEL Added

### LINKED ISSUE #485: GETDEL command support
[GETDEL](https://redis.io/commands/getdel) documentation

### PR DESCRIPTION
**Pull request by arpandaze**
This PR adds the following commands:
1. [GETEX](https://redis.io/commands/getex) (Available since 6.2.0)
2. [GETDEL](https://redis.io/commands/getdel) (Available since 6.2.0)

Closes #485

### CODE REVIEW

**Comment by arpandaze** [comment]
Hi @djc ! Can you please approve this?

File: src/commands.rs [hunk 1]
```diff
@@ -397,6 +397,16 @@ implement_commands! {
         cmd("PTTL").arg(key)
     }
 
+    /// Get the value of a key and set expiration
+    fn get_ex<K: ToRedisArgs>(key: K, seconds: usize) {
```
[the remark this answers is not in the document]
Comment (djc) [reply src/commands.rs]: Yeah, the MSRV issue is https://github.com/redis-rs/redis-rs/issues/624.

File: src/types.rs [hunk 2]
```diff
@@ -25,6 +25,20 @@ macro_rules! invalid_type_error_inner {
     };
 }
 
+/// Helper enum that is used to define expiry time
+pub enum ExpiryOption {
```
Comment (djc) [inline src/types.rs:29]: Would prefer to name this `Expiry`.

File: src/commands.rs [hunk 3]
```diff
@@ -398,8 +398,16 @@ implement_commands! {
     }
 
     /// Get the value of a key and set expiration
-    fn get_ex<K: ToRedisArgs>(key: K, seconds: usize) {
-        cmd("GETEX").arg(key).arg("EX").arg(seconds)
+    fn get_ex<K: ToRedisArgs>(key: K, expire_at: ExpiryOption) {
+        let (option, time_arg): (_, Option<usize>) = match expire_at {
```
Comment (djc) [inline src/commands.rs:402]: Is the type annotation necessary here? I would expect it isn't...

File: src/lib.rs [hunk 4]
```diff
@@ -372,6 +372,7 @@ pub use crate::pipeline::Pipeline;
 #[cfg_attr(docsrs, doc(cfg(feature = "script")))]
 pub use crate::script::{Script, ScriptInvocation};
 
+#[rustfmt::skip]
```
[the remark this answers is not in the document]
Comment (arpandaze) [reply src/lib.rs:375]: Rustfmt was changing the order of imports into this:
```
pub use crate::types::{
    // utility functions
    from_redis_value,

    // error kinds
    ErrorKind,

    ExpiryOption,

    // conversion traits
    FromRedisValue,

    // utility types
    InfoDict,
    NumericBehavior,
    // error and result types
    RedisError,
    RedisResult,
    RedisWrite,
    ToRedisArgs,

    // low level values
    Value,
};
```
I wanted to put ExpiryOption in the utility types section but rustfmt was moving it above and also removes spacing between sections.

File: src/types.rs (hunk 2 above)
Comment (arpandaze) [reply src/types.rs:29]: Sure!

File: src/types.rs [hunk 5]
```diff
@@ -25,6 +25,20 @@ macro_rules! invalid_type_error_inner {
...
+    /// EX seconds -- Set the specified expire time, in seconds.
+    EX(usize),
```
[the remark this answers is not in the document]
Comment (arpandaze) [reply src/types.rs:31]: All the other time related functions like expire, expire_at etc. used `usize` so, to make it consistent, I put it as usize.

File: src/lib.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (djc) [reply src/lib.rs:375]: Okay, maybe add a comment after the attribute like `// preserve grouping and order`

File: src/types.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (djc) [reply src/types.rs:31]: Fair enough.

File: src/commands.rs (hunk 3 above)
Comment (arpandaze) [reply src/commands.rs:402]: It isn't needed. I forgot to remove that. I will update that.

File: src/lib.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (arpandaze) [reply src/lib.rs:376]: Done!

File: src/commands.rs (hunk 3 above)
Comment (arpandaze) [reply src/commands.rs]: Done!

**Comment by arpandaze** [comment]
I have made the requested changes.

**Pull request merged**